### PR TITLE
[TKAI-3585] Check unused locales

### PR DESCRIPTION
**Pull request by RTAndrew**

### CODE REVIEW

File: lib/locales/br/frontend/api-messages.json [hunk 1]
```diff
@@ -531,8 +531,5 @@
   "E97": "Falha ao criar categoria do desafio",
   "E98": "Falha na atualização do desafio",
   "E99": "Falha na marcação da atualização do desafio",
-  "E610": "Falha ao seguir {{entity}}",
-  "E611": "Não permitido seguir este(a) {{entity}}",
-  "E612": "Falha ao deixar de seguir este(a) {{entity}}",
```
Comment (JoseMarshall) [inline lib/locales/br/frontend/api-messages.json:536]: Maybe you should ignore the api-messages, cause these codes are emitted by the API

File: package.json [hunk 2]
```diff
@@ -39,6 +40,7 @@
   },
   "homepage": "https://github.com/layerx-labs/taikai-locales#readme",
   "dependencies": {
+    "globby": "^14.0.1",
```
Comment (JoseMarshall) [inline package.json:43]: prefer to install as dev-dependency cause the lib itself doesn't depend on it.

File: scripts/check-unused-translations.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    const messagesToDeleteMap: Record<string, number> = { ...fileWithMessages['default'] };
+
+    removeKeys(messagesToDeleteMap, messagesToDeleteFromLocales);
+
+    fs.writeFile(filePath, JSON.stringify(messagesToDeleteMap, null, 2), (err) => {
+      if (err) console.log('error', err);
+    });
+  }
+};
+(async () => await extractKeys())();
```
Comment (JoseMarshall) [inline scripts/check-unused-translations.ts:63]: I'm not sure if I understood the purpose of this script

File: lib/locales/br/frontend/api-messages.json (hunk 1 above)
Comment (RTAndrew) [reply lib/locales/br/frontend/api-messages.json:536]: Thnak you. I also included the `countries`

File: scripts/check-unused-translations.ts (hunk 3 above)
Comment (RTAndrew) [reply scripts/check-unused-translations.ts]: The frontend runs a check of unused messages and returns a file with an array of the message paths that must be deleted from the locales (or not being used)
```
export const messagesToDeleteFromLocales = [
  'report.report_project',
  'report.reason.report_reason_content',
  'report.reason.report_reason_spam',
  'connected_accounts.connect_metamask_wallet_description',
];
```

Then, the `checkUnusedTranslationMessages` script deeply deletes those message paths

**Review by ickas — APPROVED** [review]

**Pull request merged**